plugin.py

```diff
@@ -23,6 +23,66 @@ class LspAstroPlugin(NpmClientHandler):
         "nodeServer.js",
     )
 
+    typescript_relpath = os.path.join(
+        'node_modules',
+        'typescript',
+        'lib',
+        'tsserverlibrary.js'
+    )
+
+    @classmethod
+    def find_typescript_path(cls, path):
+        """
+        Find typescript in the package specified ``path`` belongs to.
+
+        :param path:
+            The absolute path of a directory to use as starting point
+
+        :returns:
+            The absolute path to typescript language-server if exists.
+        """
+        name = True  # to get started with loop
+        while path and name:
+            if os.path.isfile(os.path.join(path, 'package.json')):
+                typescript_path = os.path.join(path, cls.typescript_relpath)
+                if os.path.isfile(typescript_path):
+                    return typescript_path
+                break
+
+            path, name = os.path.split(path)
+
+        return None
+
+    @classmethod
+    def is_allowed_to_start(cls, window, initiating_view, workspace_folders, configuration):
+        if configuration is None:
+            return
+
+        if configuration.init_options.get('typescript.serverPath'):
+            return  # don't find the `typescript.serverPath` if it was set explicitly in LSP-astro.sublime-settings
+
+        typescript_path = None
+
+        # try to find package root and typescript based on current file to support
+        # nested packages, which are not directly added as folder to sidebar
+        if initiating_view:
+            file_name = initiating_view.file_name()
+            if file_name:
+                typescript_path = cls.find_typescript_path(os.path.dirname(file_name))
+
+        # try to find typescript from first workspace folder
+        if workspace_folders and not typescript_path:
+            typescript_path = cls.find_typescript_path(workspace_folders[0].path)
```

Comment:
I would give workspace folders priority over current file. Any particular reason why you did it this way?

I suppose you could find corner cases where one would perform better over another but workspace folder based approach seems more stable.

Reply:
The reason is simple. 

Considder 

1. several folders being added to sidebar each of them a separate Astro package, maybe each of them using a different version of Typescript.
2. a folder being added to sidebar which contains many sub-folders each of them being an Astro package.
3. a user opens a astro file which is part of an Astro package without having its package (or any) folder added to sidebar.

In 1.) all files would use typescript version of 1st workspace folder, which may not be true.
In 2.) and 3.) no package specific typescript would be found at all. ST's default build system suffers this issue for instance, when using "key files".

Giving a view precedence just makes sure the package the file belongs to is identified reliably.